src/models/errors.ts

```diff
@@ -37,3 +37,8 @@ export class MissingConsentFieldsError extends Error {
     this.consent = consent
   }
 }
+export class InvalidConsentStatusError extends Error {
+  public constructor(consentStatus: string) {
```

Comment:
I suggest adding also as a parameter the consent.id so you can attach it to the error message which will be helpful later with debugging